src/main/java/ch/njol/skript/classes/data/SkriptClasses.java

```diff
@@ -721,6 +725,54 @@ public String toVariableNameString(final Script script) {
 						return path.relativize(file.toPath().toAbsolutePath()).toString();
 					}
 				}));
+
+		Classes.registerClass(new ClassInfo<>(Executable.class, "executable")
+			.user("executables?")
+			.name("Executable")
+			.description("Something that can be executed (run) and may accept arguments, e.g. a function.",
+					"This may also return a result.")
+			.usage("")
+			.examples("run {_function} with arguments 1 and true")
+			.since("INSERT VERSION"));
+
+		Classes.registerClass(new ClassInfo<>(DynamicFunctionReference.class, "function")
+			.user("functions?")
+			.name("Function")
+			.description("A function loaded by Skript.",
+					"This can be executed (with arguments) and may return a result.")
+			.usage("")
+			.examples("run {_function} with arguments 1 and true",
+					"set {_result} to the result of {_function}")
+			.since("INSERT VERSION")
+			.parser(new Parser<DynamicFunctionReference<?>>() {
```

Comment:
can this be a named inner class

Reply:
I'm not sure what you mean by this.

Reply:
Like `private static class DynamicFunctionReferenceParser extends Parser...`. I did this in the JavaClasses rework to make the registration section easier to navigate, however its not that big of a deal as most parsers are just an anonymous class

Reply:
I see. I'm not sure this would add a huge amount of value, since it would just be moved to another place, not made any smaller.